compiler/rustc_codegen_ssa/src/back/link.rs

```diff
@@ -559,6 +559,9 @@ fn link_staticlib<'a>(
             archive_builder_builder
                 .extract_bundled_libs(path, tempdir.as_ref(), &relevant_libs)
                 .unwrap_or_else(|e| sess.dcx().emit_fatal(e));
+
+            // The order in which we add archives is not important
+            #[allow(rustc::potential_query_instability)]
```

Comment:
Is this true? It would be nice if there was a short explanation why it is not relevant
(or just a change to sort them instead, which is easier to prove correct)

Reply:
Well, in my understanding, archives are basically directories, so mappings from file path to file. So it should really matter in which order we add the files to the archive. That being said, perhaps it would be inded better to sort, to make sure that the resulting archive will be bit-by-bit identical.

Reply:
Yes, I would prefer sorting over hopeful `allows`.

Reply:
```rust
 let mut relevant_libs: Vec<Symbol> = relevant_libs.into_iter().collect();
relevant_libs.sort_unstable();
```
produces
```
   --> compiler/rustc_codegen_ssa/src/back/link.rs:563:64
    |
563 |             let mut relevant_libs: Vec<Symbol> = relevant_libs.into_iter().collect();
    |                                                                ^^^^^^^^^
    |
    = note: if you believe this case to be fine, allow this lint and add a comment explaining your rationale
    = note: `-W rustc::potential-query-instability` implied by `-W rustc::internal`
    = help: to override `-W rustc::internal` add `#[allow(rustc::potential_query_instability)]`
```
How can I even sort it without the lint firing? :)

Reply:
Allowing it on the first statement is fine, the sort below is obvious.

Reply:
Ok, done.